src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,114 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private final int capacity = 10;
+    private final int changeStep = 1;
+    private final int zeroIndex = 0;
+    private T[] userArray;
+    private int size;
+
+    public ArrayList() {
+        userArray = (T[]) new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
-
+        changeCapacity(size + changeStep);
+        userArray[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < zeroIndex || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index of element");
+        }
+        changeCapacity(size + changeStep);
+        System.arraycopy(userArray, index, userArray, index + changeStep, size - index);
+        userArray[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < zeroIndex || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index of element");
+        } else {
+            return userArray[index];
+        }
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < zeroIndex || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("This index not exist");
+        }
+        userArray[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < zeroIndex || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index of element");
+        }
```

Comment:
repeated code,extract to a private method, include index and size in the message. You can hardcode 0 value for zeroIndex